test/e2e/cht-docker-utils.js

```diff
@@ -0,0 +1,134 @@
+const path = require('path');
+const fs = require('fs');
+const https = require('https');
+const { spawn } = require('child_process');
+const request = require('request-promise-native');
+
+const log = require('../../src/lib/log');
+
+const DEFAULT_PROJECT_NAME = 'cht_conf_e2e_tests';
+const dockerHelperDirectory = path.resolve(__dirname, '.cht-docker-helper');
+const dockerHelperScript = path.resolve(dockerHelperDirectory, './cht-docker-compose.sh');
+
+const downloadDockerHelperScript = () => new Promise((resolve, reject) => {
+  const file = fs.createWriteStream(dockerHelperScript, { mode: 0o755 });
+  https
+    // TODO: switch back to using `master` branch of cht-core once DNS issue is resolved - https://github.com/medic/medic-infrastructure/issues/571#issuecomment-2209120441
+    .get('https://raw.githubusercontent.com/medic/cht-core/dnm-docker-helper-experiments/scripts/docker-helper-4.x/cht-docker-compose.sh', (response) => {
+      response.pipe(file);
+      file.on('finish', () => file.close(resolve));
+      file.on('error', () => file.close(reject));
+    })
+    .on('error', () => {
+      fs.unlinkSync(file.path);
+      file.close(() => reject('Failed to download CHT Docker Helper script "cht-docker-compose.sh"'));
+    });
+});
+
+const ensureScriptExists = async () => {
+  if (!fs.existsSync(dockerHelperDirectory)) {
+    await fs.promises.mkdir(dockerHelperDirectory);
+  }
+
+  if (!fs.existsSync(dockerHelperScript)) {
```

Comment:
An alternative would be to maybe cook up some kind of npm `postinstall` script that re-downloads the script any time `npm ci` or `npm install` is run.  (Though I would want to be sure the `postinstall` only gets executed on the dev workflows and not when an app developer is just trying to globally install cht-conf.

Reply:
I agree with the removal of `.cht-docker-helper` at the end of the teardown but I'd rather keep those checks here because someoen could potentially cancel the test run and end up not running the `tearDownCht` function, resulting in a running cht instance and leftover files on the system.
That way we keep running `tearDownCht` in the `before` hook and clean up any leftovers